src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Definition/DefinitionEndpoint.cs

```diff
@@ -88,18 +88,20 @@ public void ApplyCapabilities(VSInternalServerCapabilities serverCapabilities, V
             positionInfo.LanguageKind));
     }
 
-    protected async override Task<DefinitionResult?> HandleDelegatedResponseAsync(
-        DefinitionResult? response,
+    protected async override Task<DefinitionResult> HandleDelegatedResponseAsync(
+        DefinitionResult response,
         TextDocumentPositionParams originalRequest,
         RazorRequestContext requestContext,
         DocumentPositionInfo positionInfo,
         CancellationToken cancellationToken)
     {
-        if (response is not DefinitionResult result)
+        if (response is null)
         {
             return null;
         }
 
+        var result = response.Value;
```

Comment:
Nit: The old code was a _teeny little bit_ faster. The `.Value` call does a check for a real value, so it can throw if it's not there.

![CantSeeAnythingKenJeongGIF](https://github.com/user-attachments/assets/8bc98734-c167-4e78-8546-1af1b870507f)